vault_core/src/vault.rs

```diff
@@ -822,6 +824,89 @@ impl Vault {
         })
     }
 
+    /// Checks to see if the minimum amount out is acceptable
+    pub fn check_min_amount_out(
+        &self,
+        amount_in: u64,
```

Comment:
can you add units here? also should check_min_amount_out use calculate_min_amount_iout?